File: UI/src/app/ticket.ts
import { TicketStatus } from "./enums/ticket.enum";

export interface Ticket {
    ticketId: string;
    product: string;
    problemDescription: string;
    status: TicketStatus;
    createdDate: Date;
    assignedToUserName: string;
    assignedToFullName: string;
    sequentialId: number;
    comments?: string[];  // Add this property
    attachments?: { attachmentId: string, fileName: string, fileUrl: string }[];  // Add this property
}

File: APIs/CSTS.DAL/Models/Ticket.cs
using CSTS.DAL.Enum;
namespace CSTS.DAL.Models {
    public class Ticket
    {
        public Guid TicketId { get; set; } // Primary Key
        public string Product { get; set; }
        public string ProblemDescription { get; set; }
        public TicketStatus Status { get; set; } // Enum
        public DateTime CreatedDate { get; set; }
        public DateTime ModifiedDate { get; set; }

        public Guid CreatedById { get; set; } // Foreign Key to User
        public User CreatedBy { get; set; }

        public Guid? AssignedToId { get; set; } // Foreign Key to User, Nullable
        public User? AssignedTo { get; set; }

        public ICollection<Comment>? Comments { get; set; }
        public ICollection<Attachment>? Attachments { get; set; } // Updated

        public Ticket()
        {
            Comments = new List<Comment>();
            Attachments = new List<Attachment>(); // Initialize the Attachments collection
        }
    }
}
